### ISSUE TITLE
lint: unnamedResult should not warn on different named types

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#259]
Reproducer:
```
gocritic check-package -enable=unnamedResult io
```
Output:
```
io/pipe.go:186:1: unnamedResult: consider to give name to results
```
Code:
```go
func Pipe() (*PipeReader, *PipeWriter) {
```
In my opinion, this signature is perfectly fine.
If there are distinct named types, type names serve as a "concise names".

Negative test suite should be extended as well.

**Comment by Quasilyte** [comment]
This was foreseen by @fexolm. Found reproducer recently.

**Comment by cristaloleg** [comment]
So..what is difference between `func f() (float64, float32)` ? 🤔 
Why we cannot add `r` and `w` for example? (might be a bit artificial, but totally harmless)

**Comment by Quasilyte** [comment]
Because it will clutter the godoc.
Also, we should be as close to https://github.com/golang/go/wiki/codereviewcomments#named-result-parameters as possible, and it does only mention more than 1 result of the same type.

**Comment by Quasilyte** [comment]
Named params help in case of general types like `string` or `int`, but doesn't add anything to named types, if there are no duplicates (because otherwise we need clarification).

**Comment by cristaloleg** [comment]
> help in case of general types
Indeed. We might limit checker only to basic types.
Does it look like a solution for you?

**Comment by cristaloleg** [comment]
Yep, sounds good. 
Yell on same types without names or on unnamed basic types.
And skip `(T, bool)` & `(T, error)` patterns as before.
So as you've described in quotes 😄 

BTW we've this issue https://github.com/golang/go/issues/20859

**Comment by Quasilyte** [comment]
> BTW we've this issue golang/go#20859

It's still open? O_o
Thought it was solved several times now.

**Comment by Quasilyte** [comment]
@cristaloleg, are you going to fix this in near time or it's OK to lend you a hand here?
